src/lualib/Error.ts

```diff
@@ -0,0 +1,93 @@
+Error = class {
+    public static captureStackTrace(targetObject: {}, depth: number): void {
+        targetObject["stack"] = debug.traceback(undefined, depth);
+    }
+
+    public message: string;
+    public name: string;
+    public stack?: string;
+
+    constructor(message?: string, name?: string) {
+        this.message = message || "";
+        this.name = name || "Error";
+        Error["captureStackTrace"](this, name ? 5 : 4);
+    }
+
+    public __tostring(): string {
+        return `${this.name}: ${this.message}`;
```

Comment:
```suggestion
        return _VERSION === "Lua 5.1" ? `${this.name}: ${this.message}` : this.stack;
```

- Lua 5.1 doesn't use tostring on errors
- LuaJIT uses tostring and appends stack to it
- Lua 5.2 and 5.3 uses tostring and uses it's raw result

Because of that we have to show a short form for JIT and add stack for > 5.2. I'm not aware of any good way to differentiate between JIT and 5.1, but since 5.1 would require some top-level handler anyway we can just use `_VERSION === "Lua 5.1"`.

Reply:
Do we really want `tostring(error)` to emit the full stacktrace? AFAICT in JavaScript printing the error usually gives `${name}: ${message}`.

Reply:
`__tostring` there is generally added not for `tostring(error)`, but for standard top-level lua error handler, which usually attaches stack trace to thrown strings, but doesn't do that for objects in some versions. Indeed, it would be nice to make `tostring(error)` consistent as well, maybe it would be possible with some call stack inspection?

TBH at this point all of that sounds pretty complex, maybe we should consider always using `${name}: ${message}` for `__tostring` and provide some custom top-level handler to attach stack.

Reply:
I would also rather have such functionality in the top-level handler rather than the error class itself.

Reply:
Looking into this more, I don't think there is a way to have a top-level error handler on all platforms. And anyway, depending on some extra thing user has to do for a quite basic feature like stacktraces doesn't sound like a good enough user experience.

I have played around it a bit, and this seems to consistently have stacks attached when thrown (tested on Lua 5.3 and JIT), but not when stringified manually:

```ts
const caller = debug.getinfo(3)
return _VERSION === "Lua 5.1" || (caller && caller.func !== error) ? `${this.name}: ${this.message}` : this.stack;
```

Reply:
@Perryvw what are your thoughts on this?

Reply:
Thinking some more about your code, I think it's actually a good way to handle error printing.